Test/Unit/Query/EncloseNames.php

```diff
@@ -0,0 +1,273 @@
+                    ->select('b')
+                    ->from('foo')
+                    ->where('"b" > 10') // manual enclose
+                    ->groupBy('a')
+                    ->having('"a" > 42') // manual enclose
+                    ->orderBy('a')
+                    ->limit('2')
+                    ->offset('1')
+            )
+            ->then
+                ->string($result)
+                    ->isEqualTo(
+                        'SELECT COUNT("a"), "b" FROM "foo" ' .
+                        'WHERE "b" > 10 ' .
+                        'GROUP BY "a" ' .
+                        'HAVING "a" > 42 ' .
+                        'ORDER BY "a" LIMIT 2 OFFSET 1'
+                    );
+    }
+
+    public function case_select_alias()
+    {
+        $this
+            ->given($q = new CUT())
+            ->when(
+                $s = $q->select(),
+                $s->enableEncloseNames(),
+                $result = (string) $s->select('"a" AS "bar"') // manual enclose
+                    ->select('b')
+                    ->from('foo')
+                    ->where('"b" > 10') // manual enclose
+                    ->groupBy('a')
+                    ->having('"a" > 42') // manual enclose
+                    ->orderBy('a')
+                    ->limit('2')
+                    ->offset('1')
+            )
+            ->then
+                ->string($result)
+                    ->isEqualTo(
+                        'SELECT "a" AS "bar", "b" FROM "foo" ' .
+                        'WHERE "b" > 10 ' .
+                        'GROUP BY "a" ' .
+                        'HAVING "a" > 42 ' .
+                        'ORDER BY "a" LIMIT 2 OFFSET 1'
+                    );
+    }
+
+    public function case_select_join()
+    {
+        $this
+            ->given($q = new CUT())
+            ->when(
+                $s = $q->select(),
+                $s->enableEncloseNames(),
+                $result = (string) $s->select('a')
+                    ->select('b')
+                    ->from('foo')
+                    ->join('bar')
+                    ->on('"b" = "baz"') // manual enclose
+                    ->where('"b" > 10') // manual enclose
+            )
+            ->then
+                ->string($result)
+                    ->isEqualTo(
+                        'SELECT "a", "b" FROM "foo" ' .
+                        'JOIN "bar" ON "b" = "baz" ' .
+                        'WHERE "b" > 10'
+                    );
+    }
+
+    public function case_insert()
+    {
+        $this
+            ->given($q = new CUT())
+            ->when(
+                $i = $q->insert(),
+                $i->enableEncloseNames(),
+                $result = (string) $i->into('foo')
+                    ->on('a', 'b')
+                    ->values(1, 'bar')
+            )
+            ->then
+                ->string($result)
+                    ->isEqualTo(
+                        'INSERT INTO "foo" ("a", "b") VALUES (1, bar)'
```

Comment:
Not related to this PR: Is `VALUES (1, bar)` correct? `bar` must be a string no?